src/equations/ideal_glm_mhd_multiion.jl

```diff
@@ -263,4 +263,171 @@ end
 
     return SVector(cons)
 end
+
+# Specialization of DissipationEntropyStable for the multi-ion GLM-MHD equations
+# See
+# - A. Rueda-Ramírez, A. Sikstel, G. Gassner, An Entropy-Stable Discontinuous Galerkin Discretization
+#   of the Ideal Multi-Ion Magnetohydrodynamics System (2024). Journal of Computational Physics.
+#   [DOI: 10.1016/j.jcp.2024.113655](https://doi.org/10.1016/j.jcp.2024.113655).
+@inline function (dissipation::DissipationEntropyStable)(u_ll, u_rr,
+                                                         orientation_or_normal_direction,
+                                                         equations::AbstractIdealGlmMhdMultiIonEquations)
+    @unpack gammas = equations
+    λ = dissipation.max_abs_speed(u_ll, u_rr, orientation_or_normal_direction,
+                                  equations)
+
+    w_ll = cons2entropy(u_ll, equations)
+    w_rr = cons2entropy(u_rr, equations)
+    prim_ll = cons2prim(u_ll, equations)
+    prim_rr = cons2prim(u_rr, equations)
+    B1_ll, B2_ll, B3_ll = magnetic_field(u_ll, equations)
+    B1_rr, B2_rr, B3_rr = magnetic_field(u_rr, equations)
+    psi_ll = divergence_cleaning_field(u_ll, equations)
+    psi_rr = divergence_cleaning_field(u_rr, equations)
+
+    # Some global averages
+    B1_avg = 0.5f0 * (B1_ll + B1_rr)
+    B2_avg = 0.5f0 * (B2_ll + B2_rr)
+    B3_avg = 0.5f0 * (B3_ll + B3_rr)
+    psi_avg = 0.5f0 * (psi_ll + psi_rr)
+
+    dissipation = zero(MVector{nvariables(equations), eltype(u_ll)})
+
+    beta_plus_ll = 0
+    beta_plus_rr = 0
+    # Compute the dissipation for the hydrodynamic quantities of all components
```

Comment:
Maybe note that this is building the matrix terms given above equation (81) in the given reference.

Reply:
I improved the comments in 5aabeda40fc536aab7bae358ae3f5c5eccff57c3.